### PR TITLE
Add basic implementation of trace RPCs, spin up gRPC server

### PR DESCRIPTION
**Pull request by the-ericwang35**

### CODE REVIEW

**Review by mabdi3 — APPROVED** [review]
LGTM

File: cmd/mock_server.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	flag.Parse()
+
+	lis, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", *port))
+	if err != nil {
+		log.Fatalf("mock server failed to listen: %v", err)
+	}
+
+	grpcServer := grpc.NewServer()
+	cloudtrace.RegisterTraceServiceServer(grpcServer, &trace.MockTraceServer{})
+
```
Comment (tbarker25) [inline cmd/mock_server.go:47]: Can we add an info log here along the lines of: "listening on localhost:8080".

File: cmd/mock_server.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	"google.golang.org/genproto/googleapis/devtools/cloudtrace/v2"
+	"google.golang.org/grpc"
+
+	"github.com/googleinterns/cloud-operations-api-mock/server/trace"
+)
+
+const defaultPort = 8080
+
+var port = flag.Int("port", defaultPort, "The port to run the server on.")
```
Comment (tbarker25) [inline cmd/mock_server.go]: How would we configure this to listen for non-local connections?

Can we change this flag to 'address' and have it take the entire address string e.g. "localhost:8080"?

**Review by tbarker25 — APPROVED** [review]
LGTM 👍

File: cmd/mock_server.go (hunk 1 above)
Comment (the-ericwang35) [reply cmd/mock_server.go:47]: Yep, done

File: cmd/mock_server.go (hunk 2 above)
Comment (the-ericwang35) [reply cmd/mock_server.go]: Good point, made the change.

**Pull request merged**